app/src/main/java/ch/epfl/qedit/model/MatrixFormat.java

```diff
@@ -31,6 +35,36 @@ public MatrixFormat(
         this.hintString = hint();
     }
 
+    public static MatrixFormat parse(String format) {
+        /** Match format: 'matrixNxM' where N and M are [0-9]+ */
+        if (Pattern.compile("^matrix(\\d+)x(\\d+)$").matcher(format).find()) {
+            /** Extract the row and column size */
+            Matcher number = Pattern.compile("(\\d+)").matcher(format);
+            number.find();
+            int i = Integer.parseInt(number.group(1));
+            number.find();
+            int j = Integer.parseInt(number.group(1));
+            return new MatrixFormat(i, j);
+        } else {
+            return null;
```

Comment:
`return null` ? Returning an exception would be better in my humble opinion

Reply:
the return null is usefull when there is more answerformat: in the AnswerFormat.parse we check if MatrixFormat.parse  return null, if it does, we try to parse with an other answerformat, if not, we return the matrixformat